docs/tutorials/forum-application/4.query-message.md

```diff
@@ -0,0 +1,74 @@
+# Query Message
+
+**In this section you will learn how a user query messages on the forum application.**
+
+## Query
+
+The `Query` method is used to query the application state for specific information. It allows clients to request data
+from the application without submitting a transaction.
+
+The `Query` method is responsible for retrieving data from the application state based on the provided query parameters.
+The query parameters can be specified in the `QueryRequest` object, which typically includes information such as the
+query path and the query data.
+
+The application is expected to interpret the query parameters and return the corresponding data in a `QueryResponse` object.
+The response may include the requested data, error codes, or any other relevant information based on the application's implementation.
+
+The `Query` method is usually particularly useful for retrieving specific information from the application, such as
+account balances, transaction history, or any other data stored in the application's state.
+
+**Following is the code for the `Query` function:**
+
+```go
+// Query the application state for specific information
+func (app *ForumApp) Query(ctx context.Context, query *abci.QueryRequest) (*abci.QueryResponse, error) {
+	fmt.Println("Executing Application Query")
+
+	resp := abci.QueryResponse{Key: query.Data}
+
+	// Parse sender from query data
+	sender := string(query.Data)
+
+	if sender == "history" {
+		messages, err := model.FetchHistory(app.state.DB)
+		if err != nil {
+			return nil, err
+		}
+		resp.Log = messages
+		resp.Value = []byte(messages)
+
+		return &resp, nil
+	}
+	// Retrieve all message sent by the sender
+	messages, err := model.GetMessagesBySender(app.state.DB, sender)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert the messages to JSON and return as query result
+	resultBytes, err := json.Marshal(messages)
+	if err != nil {
+		return nil, err
+	}
+
+	resp.Log = string(resultBytes)
+	resp.Value = resultBytes
+
+	return &resp, nil
+}
+```
+
+**Explanation of code:**
+
+`Query` function queries the application state. It takes a context and a query as input and returns a response or an error.
+
+If the query data is equal to "history", it fetches the history of messages from the application's state database and
```

Comment:
Consider adding checks in the different ABCI calls (or in `model`) to make sure no user can be call "history"

Reply:
That's a good point, I'll add some logic to the tx validation to prevent users submitting a key `history` and that will be used in the check tx validation